src/helix-ui/elements/HXDivElement.js

```diff
@@ -33,22 +33,21 @@ export class HXDivElement extends HXElement {
         return [ 'scroll' ];
     }
 
-    // FIXME: ensure 'scroll' event listener added on connect (if scroll attr present)
-    // FIXME: ensure 'scroll' event listener is removed on disconnect
-
     $onAttributeChange (attr, oldVal, newVal) {
-        if (attr === 'scroll') {
-            // FIXME: only run if connected
+        if (attr === 'scroll' && this.isConnected) {
             if (newVal !== null) {
                 this._resetScroll();
                 this.addEventListener('scroll', onScroll);
             } else {
-                // FIXME: listener doesn't get removed if element disconnects
                 this.removeEventListener('scroll', onScroll);
             }
         }
     }
 
+    $onDisconnect () {
+        this.removeEventListener('scroll', onScroll);
+    }
+
```

Comment:
Added `$onDisconnect()` based on the comment:

> // FIXME: listener doesn't get removed if element disconnects

https://github.com/rackerlabs/helix-ui/pull/418/files#diff-f7d3735db68684c95b1432479629fe2aL46

I didn't noticed any patterns without the `$onConnect()`, so adding a note here to verify this with the team.

Reply:
```javascript
$onConnect () {
  ...
  if (this.hasAttribute('scroll')) {
    this._addScrollListeners();
  }
}
```